nodeup/pkg/model/context.go

```diff
@@ -507,16 +503,10 @@ func (c *NodeupModelContext) NodeName() (string, error) {
 }
 
 // EvaluateHostnameOverride returns the hostname after replacing some well-known placeholders
-func EvaluateHostnameOverride(hostnameOverride string) (string, error) {
-	if hostnameOverride == "" || hostnameOverride == "@hostname" {
+func EvaluateHostnameOverride(cloudProvider string) (string, error) {
+	if cloudProvider != "aws" {
 		return "", nil
```

Comment:
You should move this block in here:
````
	if nodeName == "" {
		hostname, err := os.Hostname()
		if err != nil {
			klog.Fatalf("Couldn't determine hostname: %v", err)
		}
		nodeName = hostname
	}
````
That way this function will always return the appropriate hostname and you can remove the blank check above.

Reply:
Then do we need this function? Or it can be removed?

Reply:
The only thing this does is pass in the node name to protokube, which is in turn used to query its own node labels. 

Interestingly enough, if you remove that function entirely and always set the os name, I believe that going _back_ to using the hostname annotation will make this work.

I wonder if you can actually just go into `protokube/pkg/protokube/labeler.go` and just fetch the os hostname there.

Reply:
Scratch that. Forgot that it was also used for the certname. Then we actually need to know the actual nodename. (in `buildMasterKubeletKubeconfig`)